youtube_dl/extractor/hketv.py

```diff
@@ -0,0 +1,157 @@
+        }
+        _APPS_BASE_URL = 'https://apps.hkedcity.net'
+        handler_url = _APPS_BASE_URL + '/media/play/handler.php'
+
+        response = self._download_json(
+            handler_url, video_id,
+            data=urlencode_postdata(data),
+            headers=merge_dicts({'Content-Type': 'application/x-www-form-urlencoded'},
+                                self.geo_verification_headers()))
+
+        result = response.get('result')
+
+        formats = []
+        subtitles = {}
+
+        if response.get('success') and response.get('access'):
+            width = str_to_int(result.get('width'))
+            height = str_to_int(result.get('height'))
+
+            playlist0 = try_get(result, lambda x: x['playlist'][0], dict)
+            fmts = try_get(playlist0, lambda x: x['sources'], list)
+            for fmt in fmts:
+                label = fmt.get('label')
+                if label == 'HD':
+                    h = 720
+                elif label == 'SD':
+                    h = 360
+                w = h * width // height
+                urlh = self._downloader.urlopen(_APPS_BASE_URL + fmt.get('file'))
+                formats.append({
+                    'format_id': label,
+                    'ext': fmt.get('type'),
+                    'url': urlh.geturl(),
+                    'width': w,
+                    'height': h,
+                })
+
+            tracks = try_get(playlist0, lambda x: x['tracks'], list)
+            for track in tracks:
+                if not isinstance(track, dict):
+                    continue
+                track_kind = track.get('kind')
+                if not track_kind or not isinstance(track_kind, compat_str):
+                    continue
+                if track_kind.lower() not in ('captions', 'subtitles'):
+                    continue
+                track_url = _APPS_BASE_URL + track.get('file')
+                if not track_url:
+                    continue
+                track_label = track.get('label')
+                subtitles.setdefault(self._CC_LANGS.get(track_label, track_label), []).append({
+                    'url': self._proto_relative_url(track_url),
+                    'ext': 'srt',
+                })
+
+        else:
+            error = clean_html(response.get('access_err_msg'))
+            if 'Video streaming is not available in your country' in error:
+                raise GeoRestrictedError(error)
+            else:
+                raise ExtractorError(error)
+
+        # Likes
+        emotion = self._download_json(
+            'https://emocounter.hkedcity.net/handler.php',
+            video_id,
+            data=urlencode_postdata({
+                'action': 'get_emotion',
+                'data[bucket_id]': 'etv',
+                'data[identifier]': video_id,
+            }),
+            headers={'Content-Type': 'application/x-www-form-urlencoded'},
+            fatal=False)
+        if emotion.get('result'):
```

Comment:
`emotion` is not guaranteed to be a dict.

Reply:
Thanks!  Line removed.

Reply:
> Thanks!  Line removed.

As in
```python
    like_count = int_or_none(try_get(emotion, lambda x: x['data']['emotion_data'][0]['count']))
```
alone will work just fine without the extraneous `if emotion.get('result'):` check.